fastestimator/op/numpyop/multivariate/read_mat.py

```diff
@@ -38,16 +39,30 @@ class ReadMat(NumpyOp):
         parent_path: Parent path that will be prepended to a given filepath.
     """
     def __init__(self,
-                 file: str,
-                 keys: Union[str, Iterable[str]],
+                 inputs: str,
+                 outputs: Union[str, Iterable[str]],
+                 mat_keys: Union[None, str, Iterable[str]] = None,
                  mode: Union[None, str, Iterable[str]] = None,
                  ds_id: Union[None, str, Iterable[str]] = None,
                  parent_path: str = ""):
-        super().__init__(inputs=file, outputs=keys, mode=mode, ds_id=ds_id)
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode, ds_id=ds_id)
         self.parent_path = parent_path
+
+        if mat_keys is None:
+            self.mat_keys = self.outputs
+        else:
+            self.mat_keys = mat_keys
+
         self.out_list = True
 
+        if isinstance(self.mat_keys, List) and isinstance(self.outputs, List):
+            assert len(self.mat_keys) == len(self.outputs), "keys and Output lengths must match"
+
+        if not isinstance(self.mat_keys, List):
```

Comment:
Since doing this anyways, move it above the is instance check earlier and use to_list method instead of [] (handles None better)